### ISSUE TITLE
FN useless_format: write!(f,"{}",format!())

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#3155]
````rust
        write!(
            f,
            "{}",
            format!(
                "Cargo cache '{}/':\n\n",
                &self.cargo_cache_paths.cargo_home.display()
            )
        )?;
````
should just be
````rust
        write!(
            f,
            "Cargo cache '{}/':\n\n",
            &self.cargo_cache_paths.cargo_home.display()
        )?;
````
but clippy doesn't warn apparently.
````clippy 0.0.212 (cafef7b 2018-09-06)````

**Comment by flip1995** [comment]
This also applies to `print!()`, `println!()` and `writeln!()`.

This could either be a new lint and would fit in `clippy_lints/src/write.rs` or an extension of the `USELESS_FORMAT` lint in `clippy_lints/src/format.rs`. Whatever seems easier to do.

**Comment by camsteffen** [comment]
Added in #7743
